### ISSUE TITLE
VS crashes with possibly invalid [MemberNotNull] annotation

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#44049]
**Version Used**: 

```
Microsoft Visual Studio Professional 2019 Preview
Version 16.6.0 Preview 6.0
VisualStudio.16.Preview/16.6.0-pre.6.0+30105.148
Microsoft .NET Framework
Version 4.8.03752
```

**Steps to Reproduce**:

Paste the following code into VS.
```
using System.Diagnostics.CodeAnalysis;

class C
{
  public string? field;
  
  [MemberNotNull("field")]
  public static void M()
  {
  }
  
  public void Test()
  {
    M();
    field.ToString();
  }
}

namespace System.Diagnostics.CodeAnalysis
{
  [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, AllowMultiple = true)]
  public sealed class MemberNotNullAttribute : Attribute
  {
      public MemberNotNullAttribute(params string[] members) { }
      public MemberNotNullAttribute(string member) { }
  }
}

```

**Expected Behavior**:
No crash. Probably some warnings

**Actual Behavior**:
VS crashes

**Notes**
The following test currently crashes in Roslyn's master branch with a `NullReferenceException`

```
commit 27654b066e600a350706a13ebd67d1da542c1a7b
Merge: 436d17816f6 0c9f90368a5
Author: msftbot[bot] <48340428+msftbot[bot]@users.noreply.github.com>
Date:   Thu May 7 05:33:48 2020 +0000

    Merge pull request #44029 from dotnet/merges/release/dev16.7-preview1-to-master

    Merge release/dev16.7-preview1 to master
```

```
        [Fact]
        public void Test()
        {
            var source = @"using System.Diagnostics.CodeAnalysis;

class C
{
  public string? field;
  
  [MemberNotNull(""field"")]
  public static void M()
  {
  }
  
  public void Test()
  {
    M();
    field.ToString();
  }
}";
            var comp = CreateCompilation(
                new[]
                {
                    source, MemberNotNullAttributeDefinition
                },
                options: WithNonNullTypesTrue(),
                targetFramework: TargetFramework.NetCoreApp30,
                parseOptions: TestOptions.RegularPreview);
            comp.VerifyDiagnostics();
        }
```

```
System.NullReferenceException: Object reference not set to an instance of an object.

System.NullReferenceException
Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.CSharp.LocalDataFlowPass`2.GetOrCreateSlot(Symbol symbol, Int32 containingSlot, Boolean forceSlotEvenIfEmpty) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\LocalDataFlowPass.cs:line 123
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.GetOrCreateSlot(Symbol symbol, Int32 containingSlot, Boolean forceSlotEvenIfEmpty) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 1275
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.<Scan>g__getSlotForFieldOrProperty|44_13(Symbol member) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 718
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.<Scan>g__makeMemberMaybeNull|44_7(MethodSymbol method, String memberName) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 594
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.<Scan>g__makeMembersMaybeNull|44_6(MethodSymbol method, ImmutableArray`1 members) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 585
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.<Scan>g__makeNotNullMembersMaybeNull|44_5() in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 572
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.Scan(Boolean& badRegion) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 436
   at Microsoft.CodeAnalysis.CSharp.AbstractFlowPass`2.Analyze(Boolean& badRegion, Optional`1 initialState) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\AbstractFlowPass.cs:line 435
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.Analyze(NullableWalker walker, Symbol symbol, DiagnosticBag diagnostics, VariableState initialState, Builder snapshotBuilderOpt) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 971
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.Analyze(CSharpCompilation compilation, Symbol symbol, BoundNode node, Binder binder, Conversions conversions, DiagnosticBag diagnostics, Boolean useMethodSignatureParameterTypes, MethodSymbol delegateInvokeMethodOpt, VariableState initialState, Builder analyzedNullabilityMapOpt, Builder snapshotBuilderOpt, ArrayBuilder`1 returnTypesOpt) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 951
   at Microsoft.CodeAnalysis.CSharp.NullableWalker.AnalyzeAndRewrite(CSharpCompilation compilation, Symbol symbol, BoundNode node, Binder binder, DiagnosticBag diagnostics, Boolean createSnapshots, SnapshotManager& snapshotManager, ImmutableDictionary`2& remappedSymbols) in C:\Work\roslyn\src\Compilers\CSharp\Portable\FlowAnalysis\NullableWalker.cs:line 788
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.BindMethodBody(MethodSymbol method, TypeCompilationState compilationState, DiagnosticBag diagnostics, ImportChain& importChain, Boolean& originalBodyNested, InitialState& forSemanticModel) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 1682
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethod(MethodSymbol methodSymbol, Int32 methodOrdinal, ProcessedFieldInitializers& processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 986
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 511
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.<>c__DisplayClass22_0.<CompileNamedTypeAsync>b__0() in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 401
   at Roslyn.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() in C:\Work\roslyn\src\Compilers\Core\Portable\InternalUtilities\UICultureUtilities.cs:line 141
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.WaitForWorkers() in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 316
   at Microsoft.CodeAnalysis.CSharp.MethodCompiler.CompileMethodBodies(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean emittingPdb, Boolean emitTestCoverageData, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compiler\MethodCompiler.cs:line 155
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnosticsForAllMethodBodies(DiagnosticBag diagnostics, CancellationToken cancellationToken) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compilation\CSharpCompilation.cs:line 2448
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, DiagnosticBag diagnostics, CancellationToken cancellationToken) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compilation\CSharpCompilation.cs:line 2417
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, CancellationToken cancellationToken) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compilation\CSharpCompilation.cs:line 2325
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnostics(CancellationToken cancellationToken) in C:\Work\roslyn\src\Compilers\CSharp\Portable\Compilation\CSharpCompilation.cs:line 2319
   at Microsoft.CodeAnalysis.DiagnosticExtensions.VerifyDiagnostics[TCompilation](TCompilation c, DiagnosticDescription[] expected) in C:\Work\roslyn\src\Test\Utilities\Portable\Diagnostics\DiagnosticExtensions.cs:line 413
   at 
```

**Comment by CyrusNajmabadi** [comment]
@jcouv was there any progress on this made?  Thanks!

**Comment by jcouv** [comment]
@CyrusNajmabadi No progress yet. I'm focusing on records first, then async streams issue, then some nullability attribute issues. The present issue is not super high in the stack in that latter bucket for me, because it involves someone defining their own MemberNotNull type.

**Comment by CyrusNajmabadi** [comment]
Got it.  Thanks!

**Comment by TessenR** [comment]
@jcouv 

> because it involves someone defining their own MemberNotNull type.
Why is that?
I just didn't have the latest `net5` preview with this attribute installed when I tested the crash in VS.
With the latest `net5` preview referenced the following code crashes without any custom attribute declarations:
```
using System.Diagnostics.CodeAnalysis;

class C
{
  public string? field;
  
  [MemberNotNull("field")]
  public static void M()
  {
  }
  
  public void Test()
  {
    M();
    field.ToString();
  }
}
```

**Comment by jcouv** [comment]
The scenario that crashes when the type is absent is definitely worth bumping this up. Thanks!

**Comment by kevinsun-dev** [comment]
Fixed the bug in #45563. Thanks @TessenR for reporting this issue!

**Comment by TessenR** [comment]
@kevinsun-dev Thanks for fixing this. Shouldn't Roslyn warn about such usages though as it does if I mention not existing member? According to tests in the pull request there are no warnings for the attribute in the original example. I'm not sure whether `CS8776` is a good warning id here but no warnings at all for a member name that will be ignored during the nullable analysis feels wrong.
